Source/Renderer/FramebufferManager.js

```diff
@@ -0,0 +1,126 @@
+import Framebuffer from "./Framebuffer.js";
+import PixelDatatype from "./PixelDatatype.js";
+import Renderbuffer from "./Renderbuffer.js";
+import RenderbufferFormat from "./RenderbufferFormat.js";
+import Sampler from "./Sampler.js";
+import Texture from "./Texture.js";
+import Color from "../Core/Color.js";
+import defaultValue from "../Core/defaultValue.js";
+import defined from "../Core/defined.js";
+import PixelFormat from "../Core/PixelFormat.js";
+
+/**
+ * Creates a wrapper object around a framebuffer and its resources.
+ *
+ * @private
+ * @constructor
+ */
+function FramebufferManager(options) {
+  options = defaultValue(options, defaultValue.EMPTY_OBJECT);
+  this._depthStencil = defaultValue(options.depthStencil, false);
+  this._useHdr = false;
+
+  this._framebuffer = undefined;
+  this._colorRenderbuffer = undefined;
+  this._colorTexture = undefined;
+  this._depthStencilRenderbuffer = undefined;
+  this._depthStencilTexture = undefined;
+}
+
+FramebufferManager.prototype.isDirty = function (width, height, hdr) {
+  hdr = defaultValue(hdr, false);
+  var colorTexture = this._colorTexture;
+  return (
+    !defined(colorTexture) ||
+    colorTexture.width !== width ||
+    colorTexture.height !== height ||
+    this._useHdr !== hdr
+  );
+};
+
+FramebufferManager.prototype.update = function (
+  context,
+  width,
+  height,
+  depthTexture,
+  hdr
+) {
+  depthTexture = defaultValue(depthTexture, false);
+  hdr = defaultValue(hdr, false);
+
+  if (this.isDirty(width, height, hdr)) {
+    this.destroyResources();
+    this._useHdr = hdr;
+
+    // Create color texture
+    var pixelDatatype = hdr
```

Comment:
Double check this, but I _think_ in WebGL all color attachments most have the same `internalFormat`/`format`/`type` so they would either all be floating point or all unsigned byte.

Reply:
> - Some framebuffers are constructed with multiple texture attachments, and `FramebufferManager` currently expects a single attachment for color and depth stencil textures/renderbuffers. It feels clunky to have a `textureOptions` argument to keep track of constructor options in `FramebufferManager.update`, maybe if this is needed then it could go in the constructor instead?

I think my comment partly addresses this. But see where the OIT implementation takes you. That uses multiple attachments.